std/experimental/allocator/mmap_allocator.d

```diff
@@ -66,15 +65,48 @@ struct MmapAllocator
         }
 
         /// Ditto
-        nothrow @nogc
-        bool deallocate(void[] b) shared
+        pure nothrow @nogc
+        bool deallocate(void[] b) shared const
         {
             return b.ptr is null || VirtualFree(b.ptr, 0, MEM_RELEASE) != 0;
         }
     }
 }
 
-nothrow @safe @nogc unittest
+extern (C) private @system @nogc nothrow
+{
+    ref int fakePureErrnoImpl()
+    {
+        import core.stdc.errno : errno;
+        return errno();
+    }
+}
+
+extern (C) private pure @system @nogc nothrow
```

Comment:
An explanatory comment for why it's valid to apply `pure` is necessary in the source code.

Reply:
Done.